arch/arm64/include/spinlock.h

```diff
@@ -83,6 +83,25 @@
  * -- Clear Exclusive access monitor (CLREX) This is used to
  *     clear the state of the Local Exclusive Monitor.
  */
+#if defined(CONFIG_TICKET_SPINLOCK)
+#include <stdatomic.h>
+
+/* Memory layout is related uint64_t in arm64 little endian
+ * if uint64 is 0x010203040A0B0C0D, the next is 0A0B0C0D and
+ * the owner is 01020304.
+ */
+
+struct ticket_spinlock_s
+{
+    atomic_uint next;
+    atomic_uint owner;
+};
+typedef struct ticket_spinlock_s ticket_spinlock_t;
+
+#define OWNER(l) (((struct ticket_spinlock_s *)l)->owner)
+#define NEXT(l) (((struct ticket_spinlock_s *)l)->next)
```

Comment:
OWNER/NEXT macros are too general and very easy to conflict with others since spinlock.h is included in many header file.

Reply:
You are right.
There are many place using this macro and it's easy for reading.
Maybe we can think a new name?

Reply:
you can move the macro to source file?

Reply:
Sure.

Reply:
```
# define TICKET_SPINLOCK_OWNER(l) (((struct ticket_spinlock_s *)l)->owner)
# define TICKET_SPINLOCK_NEXT(l)  (((struct ticket_spinlock_s *)l)->next)
```